### PR TITLE
Update docker publish target to hyperledger/iroha2 #1453 Fix some workflows #

### PR DESCRIPTION
**Pull request by s8sato**
Signed-off-by: s8sato <49983831+s8sato@users.noreply.github.com>

### Description of the Change
* Update docker publish target to `hyperledger/iroha2` #1453
  * Update corresponding compose images #
* Update `docker/build-push-action` to `v2` #
* Unify `runs-on: ubuntu-latest` in docker-related jobs #
* Unify `actions/upload-artifact` to `v2` #
* Update `iroha2-dev.jobs.publish-artifacts.container` to `rust:1.55-buster` #

### Possible Drawbacks
As we update both of publish target and compose source at the same time
composing may not work until images have published
so it may be better to plan the time to merge this PR

### CODE REVIEW

**Comment by s8sato** [comment]
Also changed about `iroha2` branch: the target repo and the credential

**Comment by i1i1** [comment]
Could you also correct images in `docker-compose` file(s)?

File: .github/workflows/iroha2-dev.yml [hunk 1]
```diff
@@ -64,23 +64,23 @@ jobs:
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
-          repository: ${{ secrets.DOCKERHUB_ORG }}/iroha
+          repository: hyperledger/iroha2
```
Comment (i1i1) [inline .github/workflows/iroha2-dev.yml:67]: Could you migrate action from `1.1.0` to `2.0`? It is done like that in `iroha2-dev-pr.yml`

**Comment by s8sato** [comment]
`iroha2-push-nightly.jobs.tag-nightly` is copy of
`iroha2-dev.jobs.deploy` except for
* `runs-on`
* `container`
* `steps.*.with.tags`

File: docker-compose.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 version: "3.3"
 services:
   iroha:
-    image: iroha1/iroha:iroha2-dev
+    image: hyperledger/iroha2:iroha2-dev
```
Comment (i1i1) [inline docker-compose.yml:4]: May be we can make tags like `hyperledger/iroha2:dev` `hyperledger/iroha2:cli-dev` `hyperledger/iroha2:crypto-cli-dev`

Comment (s8sato) [reply docker-compose.yml:4]: I agree. Now that the repository is dedicated to i2, the tags are natural

**Review by i1i1 — APPROVED** [review]

**Comment by eadventurous** [comment]
> `iroha2-push-nightly.jobs.tag-nightly` is copy of `iroha2-dev.jobs.deploy` except for
> 
>     * `runs-on`
> 
>     * `container`
> 
>     * `steps.*.with.tags`

In the deploy we probably also need to `runs-on: ubuntu-latest` as rust image does not have the needed Docker exec

**Review by eadventurous — CHANGES_REQUESTED** [review]
You also need to update `iroha2-dev-pr.yml` there are local docker building steps, which are needed for docker compose test.

After you update them - docker compose test will pass.

**Comment by s8sato** [comment]
* Update corresponding compose images
* Unify `runs-on: ubuntu-latest` in docker-related jobs
* Unify `actions/upload-artifact` to `v2`
* Update `iroha2-dev.jobs.publish-artifacts.container` to `rust:1.55-buster`

Also updated the top description

**Review by i1i1 — DISMISSED** [review]

**Comment by s8sato** [comment]
Remove buildx cache

**Review by i1i1 — APPROVED** [review]

**Review by eadventurous — APPROVED** [review]

**Pull request merged**